core/src/sumeragi/main_loop.rs

```diff
@@ -906,7 +966,11 @@ pub(crate) fn run<F: FaultInjection>(
                                 BlockCommitted::from(committed_block.clone()),
                             );
 
-                            sumeragi.broadcast_packet(msg, current_topology);
+                            if is_genesis_peer && sumeragi.debug_force_soft_fork {
+                                std::thread::sleep(sumeragi.pipeline_time() * 2);
```

Comment:
How reliable is this for generating a soft fork on weaker machines?

Reply:
As far as I can tell 100 %. It produces the exact scenario mentioned in the whitepaper. Proxy tail commits the block locally but goes offline for some time while the rest of the network move on and produce a new block. The new block is guaranteed to happen within the pipeline time so when the peer rejoins it is in conflict with the rest of the network.